File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def GCD(a, b):
    if a % b == 0:
        return b
    return GCD(b, a % b)

A, B = map(int, input().split())
if A > B:
    gcd = GCD(A, B)
else:
    gcd = GCD(B, A)

lcm = (A * B) // gcd

print(gcd)
print(lcm)

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int A = sc.nextInt();
        int B = sc.nextInt();
        int gcd;
        int lcm;
        if (A > B) {
            gcd = getGcd(A, B);
        } else {
            gcd = getGcd(B, A);
        }
        
        lcm = (A * B) / gcd;
            
        System.out.println(gcd);
        System.out.println(lcm);
    }
    
    static int getGcd(int a, int b) {
        if (a % b == 0) {
            return b;
        }  
        return getGcd(b, a % b);
    }
}